### PR TITLE
Issue 8724

### PR DESCRIPTION
**Pull request by ghost**
fix issue #8724 
Added test for composite numbers in factorial module

### CODE REVIEW

**Comment by pelegm** [comment]
Can you squash and rebase these commits?

**Comment by aktech** [comment]
x < 3  : doesn't makes any sense, you are comparing a sympy object with an integer, It will return "symbolic boolean expression has no truth value." and 
what if x.is_integer and x.is_nonnegative is True and nested if returns None?
then else will be executed, which is wrong!

File: sympy/functions/combinatorial/tests/test_comb_factorials.py [hunk 1]
```diff
@@ -93,6 +93,8 @@ def test_factorial():
 
     assert factorial(-2) == zoo
     assert factorial(0) == 1
+    assert factorial(1) == 1
+    assert factorial(2) == 2
```
Comment (aktech) [inline sympy/functions/combinatorial/tests/test_comb_factorials.py]: What's the purpose of these?

File: sympy/functions/combinatorial/tests/test_comb_factorials.py [hunk 2]
```diff
@@ -110,7 +112,13 @@ def test_factorial():
     assert factorial(r).is_real is None
     assert factorial(s).is_real is True
     assert factorial(t).is_real is True
+
```
Comment (aktech) [inline sympy/functions/combinatorial/tests/test_comb_factorials.py]: why new line?

File: sympy/functions/combinatorial/factorials.py [hunk 3]
```diff
@@ -176,6 +176,16 @@ def _eval_is_real(self):
         if ((self.args[0].is_negative and self.args[0].is_noninteger) or
            self.args[0].is_nonnegative):
             return True
+        
+    def _eval_is_composite(self):
+        x = self.args[0]
+        if x.is_integer and x.is_nonnegative:
+            if x < 3:
```
Comment (skirpichev) [inline sympy/functions/combinatorial/factorials.py]: this can raise an exception (symbolic boolean expression has no truth value)

**Comment by pelegm** [comment]
Original author is a deleted user; I'd say let's close this PR.

**Pull request closed without merging**